pkg/utils/devlink.go

```diff
@@ -0,0 +1,74 @@
+package utils
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/vishvananda/netlink"
+)
+
+// NetlinkFunction is a type of function used by netlink to get data
+type NetlinkFunction func(string, string) (map[string]string, error)
+
+const (
+	// fwAppNameKey is a key to extract DDP name
+	fwAppNameKey = "fw.app.name"
+	pciBus       = "pci"
+)
+
+var (
+	netlinkDevlinkGetDeviceInfoByNameAsMap = netlink.DevlinkGetDeviceInfoByNameAsMap
```

Comment:
we already have a netlink provider [1] shouldnt we extend it with the new method ?

then, the rest of the functions in this file would be moved to ddp.go as they are pretty specific for ddp.

in tests we would use a mocked version of NetlinkProvider instead of defining a fake function.


[1] https://github.com/k8snetworkplumbingwg/sriov-network-device-plugin/blob/d59e6abe605a4ce0a0e7dca82afff0bfd9fb25fc/pkg/utils/netlink_provider.go#L28

Reply:
Hi @adrianchiris yes 100% agree.

I migrated the code, but have hit an issue with the mockery.
```
2024/05/23 14:26:14 internal error: package "bytes" without types was imported from "github.com/k8snetworkplumbingwg/sriov-network-device-plugin/pkg/utils"
```

I am trying to figure out the reason, but this is blocking me from making the changes, as I cannot regenerate the mocked version of the NetlinkProvider Interface. 

I also did a test of moving the code to a test project and trying to compile but still hit the same issues with mockery. 

Has anyone come across an issue like this with mockery before?

Reply:
I found this one https://github.com/argoproj/argo-workflows/issues/8251
can you try to build the mockery version please?